### ISSUE TITLE
Large Scale GAN Training for High Fidelity Natural Image Synthesis

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#24]
# Key Idea

- Scaling up GANs in terms of ***batch size*** and ***channel width*** leads to improved performance.
- ***Truncation Trick*** enabled by orthogonal regularization of G empowers ***controlled trade-of between sample fidelity and variety***.
- GANs trained on huge dataset, e.g. JFT-300M , are safer from ***training instability***, but for now, instabilities are inevitable and thus should be delayed as much as possible, for the best.

# Previous Approaches

- To remedy training instability issue, two approaches suggested:

    (1) Modifying the Objective Function (e.g. WGAN)

    (2) Regularizing D (e.g. Spectral Normalization)

- Conditional models provide class information to generate class-conditioned sample.

    (1) Class One-hot encoding + Concatenate to latent vector

    (2) Class info injected to Batch Norm layers

    (3) Class embeddings provided to D so that G has to match the class o.w. caught

- Quantitative Evaluation of GANs is difficult, but two methods mostly used:

    (1) Inception Score: gauges fidelity + variety across classes, but not within the class.

![image](https://user-images.githubusercontent.com/46308407/104118850-3aecba00-536f-11eb-9645-a1df667fa9b9.png)

    (2) Frechet Inception Distance: uses intermediate layer output of inception network

![image](https://user-images.githubusercontent.com/46308407/104118855-40e29b00-536f-11eb-9afe-b1c0e12a4267.png)

# Formulation

### Architecture

- Self-Attention GAN (SAGAN)
- Class info injected through BN
- Spectral Normalization on G

![image](https://user-images.githubusercontent.com/46308407/104118877-5a83e280-536f-11eb-971e-bd1a965ff12d.png)

### Latent Prior

- Try diverse priors $p(z)$
- **Bernoulli** and **Censored Normal** distributions work well

### Scaling up

- Batch Size: more modes incorporated (up to 2048)
- Width: More information processed
- Depth: Significant only with skip-z connections (BigGan-deep)

### Truncation Tricks

- Truncation of prior support enables ***controlled fidelity-variety trade-off***

![image](https://user-images.githubusercontent.com/46308407/104118888-640d4a80-536f-11eb-8554-d9c21215a8f1.png)

- IS doesn't require variety within class (mainly fidelity) while FID requires both fidelity and variety
- Larger models suffer saturation artifacts when fed truncated noise.

![image](https://user-images.githubusercontent.com/46308407/104118893-6bccef00-536f-11eb-9c4b-3414e6bd73e7.png)

- To prevent this, G has to become smoother. Apply ***Orthogonal Regularization***.

![image](https://user-images.githubusercontent.com/46308407/104118900-72f3fd00-536f-11eb-81c4-03e84ad5a4a2.png)

- This original form of **O.R.** is known to ****limit too much. ($W=$ weight matrix)

![image](https://user-images.githubusercontent.com/46308407/104118907-7be4ce80-536f-11eb-9328-0bf8ea784050.png)

- Now, the weight norms are free, only their cosine similarity (covariance) is regularized.

### Instabilities

- This scaled-up GANs can produce fine images with resolution up to $512 \times 512$ without progressive growing.
- However, large GANs undergo training collapse, which necessitates early stopping.
- On Generator side, ***top three singular values of Weight matrix*** presage the onset of collapse.

![image](https://user-images.githubusercontent.com/46308407/104118913-8606cd00-536f-11eb-82a1-0de432e3c7ac.png)

- However, even when these singular values were forced tight, instability still ocurred.
- On Discriminator side, ***jump in singular values*** occured at collapse. This suggests that at certain point, D received ***abruptly large gradient from G producing highly perturbing samples.***

![image](https://user-images.githubusercontent.com/46308407/104118917-8c954480-536f-11eb-8fd1-f09ba8430a5f.png)

- Natural remedy is to employ ***gradient penalty, but it severely costed performance damage***.
- [SUMMARY] Instability is caused from D receiving abrupt perturbation from G. It can be fixed with gradient penalty, but harms the performance greatly. Delay it, as far as possible.

# Result

![image](https://user-images.githubusercontent.com/46308407/104118923-99b23380-536f-11eb-87f2-035428da92a1.png)

- Increase in Batch size boosts performance greatly. (Diversity acquired)

![image](https://user-images.githubusercontent.com/46308407/104118929-a5055f00-536f-11eb-9f61-2f2e7a0b2aa8.png)

- 4 scores resulting from truncation trick (fidelity-variety trade off) with different resolutions.

![image](https://user-images.githubusercontent.com/46308407/104118935-acc50380-536f-11eb-9f32-6e1b4818dabe.png)

- Generated images of different resolutions. Class leakage in (d)

![image](https://user-images.githubusercontent.com/46308407/104118943-b3ec1180-536f-11eb-8445-c4985cd3ffe2.png)

- Ablation on JFT-300M dataset.

# Comments

- Similar feeling as when I read BERT. HUGE models and Google dominate the world.
- Truncation trick is very intuitive, but how they achieved so-called controlled fidelity-variety trade off is very intriguing.
- How they analyzed instability issues is very impressive. Singular values seem to play an important role as an indicator of training status. More study required.
- Now that we can obtain variety-fidelity curve from truncation tricks, metrics similar to AUC might be employed for fairer comparison between models.
